pkg/apis/flowcontrol/v1beta3/conversion.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta3
+
+import (
+	"fmt"
+
+	"k8s.io/api/flowcontrol/v1beta3"
+	"k8s.io/apimachinery/pkg/conversion"
+	"k8s.io/kubernetes/pkg/apis/flowcontrol"
+)
+
+func Convert_v1beta3_PriorityLevelConfiguration_To_flowcontrol_PriorityLevelConfiguration(in *v1beta3.PriorityLevelConfiguration, out *flowcontrol.PriorityLevelConfiguration, s conversion.Scope) error {
+	if err := autoConvert_v1beta3_PriorityLevelConfiguration_To_flowcontrol_PriorityLevelConfiguration(in, out, nil); err != nil {
+		return err
+	}
+
+	// during v1beta3 -> internal conversion:
+	//  - remove the roundtrip annotation for the 'NominalConcurrencyShares' field
+	//  - make sure we don't mutate the source (v1beta3) object's annotations
+	annotations, copied := dropPriorityLevelConcurrencyShareDefaultAnnotation(in.ObjectMeta.Annotations)
```

Comment:
Wouldn't it be more robust to start from `out.ObjectMeta.Annotations` than `in.ObjectMeta.Annotations`?

Same question everywhere else this pattern is followed.

Reply:
yes, we are using `out` here just to be safe (although today i don't think the covert for priority level adds/updates any annotation)

Reply:
My point is that I saw code that is NOT using `out`, it was using `in`.